<?xml version="1.0"?>

<launch>

	<!-- LAUNCH ARGUMENTS -->

	<!-- Arguments for the playback of data -->
	<arg
		name="rosbag_file"
		default="gmapping_demo/basic_localization_stage.bag"
		>
	</arg>
	<arg
		name="scan_topic"
		default="/laser_scan"
		>
	</arg>

	<!-- Arguments for graphslam-engine -->
	<!-- to be specified by the user as a command line option -->
	<arg
	    name="ini_file" >
	</arg>
	<arg
	    name="gt_file"
	    default=""
	    >
	</arg>

	<arg
	    name="NRD"
	    default="CFixedIntervalsNRD"
	    >
	</arg>
	<arg
	    name="ERD"
	    default="CICPCriteriaERD"
	    >
	</arg>
	<arg
	    name="GSO"
	    default="CLevMarqGSO"
	    >
	</arg>

    <!-- LAUNCH PARAMETERS -->

    <param name="/use_sim_time" value="false" />
    <param name="/base_scan" value="$(arg scan_topic)" />


    <!-- Parameters for the graphslam-engine application -->

    <param
        name="/graphslam_engine/filenames/ini_file"
        value="$(arg ini_file)"
        type="string"
        >
    </param>
    <param
        name="/graphslam_engine/filenames/gt_file"
        value="$(arg gt_file)"
        type="string"
        >
    </param>

    <param
        name="/graphslam_engine/deciders_optimizers/NRD"
        value="$(arg NRD)
        "
        type="string"
        >
    </param>
    <param
        name="/graphslam_engine/deciders_optimizers/ERD"
        value="$(arg ERD)"
        type="string"
        >
    </param>
    <param
        name="/graphslam_engine/deciders_optimizers/GSO"
        value="$(arg GSO)"
        type="string"
        >
    </param>

<!-- NODES TO BE LAUNCHED -->

<!-- rosbag for playing back the data -->
<node
	pkg="rosbag"
	type="play"
	name="rosbag_player"
	args="--clock $(find topic_provider)/rosbags/$(arg rosbag_file) /base_scan:=$(arg scan_topic)"
	>
</node>

<!-- Node that parses the external .ini file and publishes necessary
			variables in the ROS parameter server -->
<node
	pkg="topic_provider"
	type="ini_parser.py"
	name="ini_parser"
	output="screen"
	>
</node>

<!--[> Node that plays back the topics to a TCP stream <]-->
<!--<node-->
	<!--pkg="topic_provider"-->
	<!--type="topic_provider_node"-->
	<!--name="topic_provider_node"-->
	<!--output="screen"-->
	<!-->-->
	<!--</node>-->

<!-- Wrapper node that launches the MRPT graphslam-engine application -->
<node
    pkg="topic_provider"
    type="graphslam_engine_launcher.py"
    name="graphslam_engine_launcher"
    output="screen"
    >
    </node>


</launch>

